### PR TITLE
Make iochaos wait for toda startup (#1545)

### PR DESCRIPTION
**Pull request by AsterNighT**
### What problem does this PR solve?
 

cherry-pick #1545

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/chaosdaemon/iochaos_server.go [hunk 1]
```diff
@@ -74,28 +78,58 @@ func (s *DaemonServer) ApplyIoChaos(ctx context.Context, in *pb.ApplyIoChaosRequ
...
+	_ = client.CallContext(timeOut, &ret, "update", actions)
+	rpcError = client.CallContext(timeOut, &ret, "get_status", "ping")
+	if rpcError != nil || ret != "ok" {
+		log.Info("Starting toda takes too long or encounter an error")
+		caller.Close()
+		receiver.Close()
+		if kerr := s.killIoChaos(ctx, int64(cmd.Process.Pid), ct); kerr != nil {
+			log.Error(kerr, "kill toda failed", "request", in)
+		}
+		return nil, fmt.Errorf("Toda startup takes too long or an error occurs: %s", ret)
```
[the remark this answers is not in the document]
Comment (AsterNighT) [reply pkg/chaosdaemon/iochaos_server.go:130]: @muse-dev ignore

**Review by cwen0 — APPROVED** [review]
LGTM

**Review by fewdan — APPROVED** [review]
LGTM

**Pull request merged**